src/org/mozilla/javascript/BaseFunction.java

```diff
@@ -206,8 +215,15 @@ protected void setInstanceIdValue(int id, Object value) {
                 }
                 return;
             case Id_name:
+                if (value == NOT_FOUND) {
+                    namePropertyAttributes = -1;
+                }
+                return;
             case Id_arity:
             case Id_length:
+                if (value == NOT_FOUND) {
```

Comment:
For amity also?

Reply:
That's an obsolete property but in the interest of completeness I put that in too.